### PR TITLE
[DX-370]Add GH action to create menu.yaml from CSV files

### PR DESCRIPTION
**Pull request by yurisasuke**
## Related Issue (if applicable)

DX-370
## Description
- We want to create github action that will generate menu.yaml  file from scripts/menu-gen/pages-list.csv and scripts/menu-gen/data-bank.csv on every push.

## Types of changes

- [ ] Fixing typo (please merge to production)
- [ ] Documenting a new feature (please merge to production)
- [ ] Documentation for future release (please do not merge to production)
- [ ] Something else (please add if needs merging to production or not)

## Checklist

- [ ] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [ ] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [ ] Make sure you have started *your change* off *our latest `master`*.
- [ ] I have added a preview link.

### CODE REVIEW

**Comment by letzya** [comment]
@Keithwachira does it mean that we will get conflicts with menu.yaml? or will the commit work?

File: .github/workflows/generate_menu.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        uses: actions/checkout@v2
+      - name: Setup Hugo
+        uses: peaceiris/actions-hugo@v2
+        with:
+          hugo-version: '0.92.1'
+          extended: true
+      - name: Run script to generate  menu.yaml
+        run: |
+          make gen-menu
+      - uses: stefanzweifel/git-auto-commit-action@v4
```
Comment (letzya) [inline .github/workflows/generate_menu.yaml:20]: change to a PR `peter-evans/create-pull-request@v5`

**Review by dcs3spp — APPROVED** [review]
PR looks good to me. Should we disable pushes for menu.yaml by adding to .gitignore file and then let the GH Action push the generated menu.yaml

**Comment by letzya** [comment]
let's close this for now since we might not need it

**Pull request closed without merging**